Engine/src/Engine/ResourceManagement/ShaderGenerator.cpp

```diff
@@ -60,4 +55,50 @@ namespace Engine
 		}
 		return true;
 	}
+
+	ShaderNorms::ShaderNorms() : m_defaultShader("TextureFill")
+	{
+		addShader(m_defaultShader);
+		addShader("ColorFill");
+	}
+
+	ShaderNorms::~ShaderNorms()
+	{
+		for (auto [key, value] : m_shaders)
+		{
+			glDeleteProgram(value);
+		}
+	}
+
+	void ShaderNorms::addShader(string shaderName)
+	{
+		m_shaders.insert(pair(shaderName, ShaderGenerator(
+			ResourceManager::getInstance()->getShaderData(shaderName + ".vs").c_str(),
+			ResourceManager::getInstance()->getShaderData(shaderName + ".fs").c_str()
+		).getProgramId()));
+		GE_CORE_INFO(shaderName +" ADDED.");
+	}
+
+	//true if a new stride amount is introduced so that GL doesn't reload the same shader
+	bool ShaderNorms::queueStride(int stride)
```

Comment:
since this returns a bool I think the naming of this method should be a bit more informative. Something like isNewStride, or something to that effect

Reply:
Done.